site/src/components/Navbar.tsx

```diff
@@ -440,9 +440,7 @@ export const Navbar: VFC<NavbarProps> = ({ navbarHeight, setNavbarHeight }) => {
 
   const [displayMobileNav, setDisplayMobileNav] = useState<boolean>(false);
   const [idleScrollPosition, setIdleScrollPosition] = useState<boolean>(false);
-  const [scrollY, setScrollY] = useState<number>(
-    typeof window === "undefined" ? 0 : window.scrollY,
-  );
+  const [scrollY, setScrollY] = useState<number>(0);
```

Comment:
Just for my understanding, what's the thinking behind this? Why not initialise the state with the correct value on the first render when the page is being rendered in the browser?

Reply:
When we use SSR, the first CSR render should be identical to the SSR results. So instead of:

```ts
const [answer, setAnswer] = useState(typeof window === "undefined" ? 0 : 42);
```

it should be:

```ts
const [answer, setAnswer] = useState(0);

useEffect(() => {
  setAnswer(42);
}, []);
```

Calls to `useEffect` are ignored on the server. The page will flickr in both cases, because the first version of the rendered page is server-side-rendered and there is a time gap between html rendering and bundle loading. `useEffect` just adds an extra 10ms delay, but makes UI behaviour more stable.

If we want to start with `42` in the case above, we should think how to get that value on the server. This is not always possible, e.g. because the value depends on window size. If that’s the case, we can make the component client-only or think about ways to replace state with CSS media queries etc.